modin/experimental/engines/omnisci_on_native/test/utils.py

```diff
@@ -0,0 +1,195 @@
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import pytest
+import modin.pandas as pd
+import pandas
+import pyarrow as pa
+
+from modin.pandas.test.utils import (
+    df_equals,
+    io_ops_bad_exc,
+    eval_io as general_eval_io,
+)
+
+from modin.experimental.engines.omnisci_on_native.frame.omnisci_worker import (
+    OmnisciServer,
+)
+
+
+def eval_io(
+    fn_name,
+    comparator=df_equals,
+    cast_to_str=False,
+    check_exception_type=True,
+    raising_exceptions=io_ops_bad_exc,
+    check_kwargs_callable=True,
+    modin_warning=None,
+    md_extra_kwargs=None,
+    *args,
+    **kwargs,
+):
+    """
+    Evaluate I/O operation and do equality check after importing Modin's data to OmniSci.
+
+    Notes
+    -----
+    For parameters description please refer to ``modin.pandas.test.utils.eval_io``.
+    """
+
+    def omnisci_comparator(df1, df2):
+        """Evaluate equality comparison of the passed frames after importing the Modin's one to OmniSci."""
+        with ForceOmnisciImport(df1, df2):
+            comparator(df1, df2)
+
+    general_eval_io(
+        fn_name,
+        comparator=omnisci_comparator,
+        cast_to_str=cast_to_str,
+        check_exception_type=check_exception_type,
+        raising_exceptions=raising_exceptions,
+        check_kwargs_callable=check_kwargs_callable,
+        modin_warning=modin_warning,
+        md_extra_kwargs=md_extra_kwargs,
+        *args,
+        **kwargs,
+    )
+
+
+class ForceOmnisciImport:
+    """
+    Trigger import execution for Modin DataFrames obtained by OmniSci engine if already not.
+
+    When using as a context class also cleans up imported tables at the end of the context.
+
+    Parameters
+    ----------
+    *dfs : iterable
+        DataFrames to trigger import.
+    """
+
+    def __init__(self, *dfs):
+        self._imported_frames = []
+        for df in dfs:
+            if not isinstance(df, (pd.DataFrame, pd.Series)):
+                continue
+            df.shape  # to trigger real execution
+            if df.empty:
+                continue
+            partition = df._query_compiler._modin_frame._partitions[0][0]
+            if partition.frame_id is not None:
+                continue
+            frame = partition.get()
+            if isinstance(frame, (pandas.DataFrame, pandas.Series)):
+                frame_id = OmnisciServer().put_pandas_to_omnisci(frame)
+            elif isinstance(frame, pa.Table):
+                frame_id = OmnisciServer().put_arrow_to_omnisci(frame)
+            else:
+                raise TypeError(
+                    f"Unexpected storage format, expected pandas.DataFrame or pyarrow.Table, got: {type(frame)}."
+                )
+            partition.frame_id = frame_id
+            self._imported_frames.append((df, frame_id))
+
+    def __enter__(self, *dfs):
+        pass
```

Comment:
it should be returning `self` so things like
```python
with ForceOmnisciImport(df) as foo:
    print(foo)
```
work correctly. Not that it is required currently, but it's better to implement a general context manager protocol from the get go.

Reply:
done